src/Gauge/NavigatorWidget.cpp

```diff
@@ -0,0 +1,314 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+// Copyright The XCSoar Project
+
+#include "NavigatorWidget.hpp"
+#include "InfoBoxes/InfoBoxWindow.hpp"
+#include "Input/InputEvents.hpp"
+#include "Interface.hpp"
+#include "Look/InfoBoxLook.hpp"
+#include "Renderer/NavigatorRenderer.hpp"
+#include "Task/Points/TaskWaypoint.hpp"
+#include "Task/TaskType.hpp"
+#include "Task/Unordered/GotoTask.hpp"
+#include "UISettings.hpp"
+#include "ui/canvas/Canvas.hpp"
+#include "BackendComponents.hpp"
+#include "DataComponents.hpp"
+#include <iostream>
+
+NavigatorWindow::NavigatorWindow(
+  const NavigatorLook &_look, const TaskLook &_look_task,
+  const InfoBoxLook &_look_infobox, const bool _inverse) noexcept
+  : look(_look), look_task(_look_task), look_infobox(_look_infobox),
+    inverse(_inverse), dragging(false) {}
+
+void
+NavigatorWindow::ReadBlackboard(const AttitudeState _attitude) noexcept {
+  attitude = _attitude;
+  // Invalidate();
+}
+
+void
+NavigatorWindow::OnPaint(Canvas &canvas) noexcept {
+  if (inverse)
+    canvas.Clear(COLOR_BLACK);
+  else
+    canvas.ClearWhite();
+
+  TaskType tp{};
```

Comment:
here's the beginning of one part in which I'm not very confident.

I would say this zero-initialiser tp{} lead to tp = TaskType::NONE;
but because of assumptions, I change it to:
TaskType tp{TaskType::NONE};